internal/services/postgres/postgresql_flexible_server_resource.go

```diff
@@ -400,8 +399,16 @@ func resourcePostgresqlFlexibleServer() *pluginsdk.Resource {
 			// storage_mb size...
 			storageTiers := storageTierMappings[newMb]
 
+			newTier = ""
+
+			// set new tier only when tier is changed or when value is faster than default
+			if oldTierRaw.(string) != newTierRaw.(string) || validate.CompareAzureManagedDiskPerformance(servers.AzureManagedDiskPerformanceTiers(newTierRaw.(string)), storageTiers.DefaultTier) {
+				newTier = newTierRaw.(string)
+			}
+
 			if newTier == "" {
 				newTier = string(storageTiers.DefaultTier)
+				diff.SetNew("storage_tier", newTier)
```

Comment:
I assume that we don't need it anymore if using "IsExplicitlyNullInConfig" to fix the issue.

Reply:
Hi @neil-yechenwei, I added it, to make it visible while running the "plan" / "deploy" action.

For sure technically it's not required.
So should I remove it?

Reply:
Seems your fix also can resolve this issue.

Reply:
Seems expandArmServerStorage func also need to be updated accordingly, right?

Reply:
Yes